ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QueryStreamHandler.java

```diff
@@ -219,6 +233,10 @@ private void handleQueryPublisher(
 
       // When response is complete, publisher should be closed and query unregistered
       routingContext.response().endHandler(v -> {
+        if (endedResponse.getAndSet(true)) {
```

Comment:
super nit: let's extract this into a `close(Closeable item)` method that gets and sets, closes the underlying item and reports the metrics?